### PR TITLE
wine: update to 9.0

### PR DESCRIPTION
**Pull request by fvalasiad**
Changes in this commit besides upgrade to 9.0:

1. The link to the source code temporarily changed since for release 9.0 they did not follow the 9.x/<release>.tar.gz convention and instead released it directly under 9.0/ . I expect them to move back to 9.x though sooner or later when 9.1 is to be released. We can change it back at that time.

2. libwine, previously marked deprecated was removed, I as a result also remove the respective package.

Additional notes include the new WoW64 functionality which no longer requires a separate wine 32bit install to run 32bit windows software, it's all to be managed by the 64bit installation, really cool, but still somewhat experimental with some drawbacks, not anything extremely annoying though, can be enabled with a command line flag.

Also it's now possible to run x86 windows apps on ARM AARCH64! Once again you need to supply a command line flag to enable it. Bright future since it now means aarch64 void linux can ship wine to arm installs too.

I haven't taken any measures to enabling any of the two, just thought I'd bring them over to your attention and ask you what you wanna do about it.

Full notes: https://gitlab.winehq.org/wine/wine/-/releases/wine-9.0

#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

File: srcpkgs/wine/template [hunk 1]
```diff
@@ -52,11 +52,11 @@ makedepends="gettext-devel ncurses-devel glu-devel libSM-devel
  SDL2-devel"
 depends="libXi libXinerama libXcomposite libXcursor libOSMesa
  desktop-file-utils hicolor-icon-theme liberation-fonts-ttf
- gnutls SDL2 wine-common>=${version}_${revision} libwine>=${version}_${revision}"
+ gnutls SDL2 wine-common>=${version}_${revision}
```
Comment (Hoshpak) [inline srcpkgs/wine/template:55]: The closing quotation mark is missing here.

File: srcpkgs/wine/template [hunk 2]
```diff
@@ -52,11 +52,11 @@ makedepends="gettext-devel ncurses-devel glu-devel libSM-devel
...
 lib32depends="libXi-32bit>=0 libXinerama-32bit>=0 libXcomposite-32bit>=0
  libXcursor-32bit>=0 libOSMesa-32bit>=0 desktop-file-utils>=0 hicolor-icon-theme>=0
  liberation-fonts-ttf>=0 gnutls-32bit>=0 SDL2-32bit>=0
- wine-common>=${version}_${revision} libwine-32bit>=${version}_${revision}"
+ wine-common>=${version}_${revision}
```
Comment (Hoshpak) [inline srcpkgs/wine/template:59]: here also

**Review by Hoshpak — CHANGES_REQUESTED** [review]

File: srcpkgs/wine/template (hunk 2 above)
Comment (fvalasiad) [reply srcpkgs/wine/template]: done

**Comment by fvalasiad** [comment]
@Hoshpak any idea about the conflicts mentioned with libwine? Is it because of the fact that srcpkgs/libwine is still there? Should I remove it in this PR as well or leave it for another one?

**Comment by Hoshpak** [comment]
The new WoW64 mode looks interesting however it seems like activating it would compromise too much on compatibility and performance so I would prefer to keep building the old way for now and activate the new mode only after it has been improved in a future version or if we wanted to drop i686 and multilib support.

Building for aarch64 would certainly be useful for some people however I don't own any aarch64 hardware right now so I couldn't test the resulting build myself.

**Comment by fvalasiad** [comment]
> The new WoW64 mode looks interesting however it seems like activating it would compromise too much on compatibility and performance so I would prefer to keep building the old way for now and activate the new mode only after it has been improved in a future version or if we wanted to drop i686 and multilib support.
> 
> Building for aarch64 would certainly be useful for some people however I don't own any aarch64 hardware right now so I couldn't test the resulting build myself.

I have a couple of SBCs and PIs around, want me to run a quick void install on em and try out wine with some basic x86 windows programs?

I am not that familiar with configuring the template for other architectures though, I will need some time to figure out how to make a package. Otherwise I cross compile wine normally and simply move the bundle to my pi and test run it there, then we invest in the arm package.

**Comment by fvalasiad** [comment]
@AlexKurisu Hmmm indeed there is no performance penalty to pay between using the old WoW64 with 32bit binaries and using the new one with only 64bit binaries. So wine on musl so far was not able to run windows 32bit apps at all? This means that the musl experience is also improved, god I love the new wine release!

I upvote, it's a good idea, what do you think @Hoshpak.

**Comment by AlexKurisu** [comment]
Just to note: Alpine builds wine with new WoW64 since it's introduction

**Comment by Hoshpak** [comment]
@fvalasiad Regarding the libwine package: Please also remove the symlink. The package should probably also declare a "replaces=libwine>=0" to ensure that it is removed during the installation of the new wine version. I think that should solve the failing check.

Getting a package to cross-compile that isn't intended to be (I'm not sure if that's actually the case for wine) can be quite challenging so I would suggest that we give it a try and do if it's a quick wine but regard it as optional and postpone it if it's not to not delay the update unnecessarily. I would have some time on the weekend to try to adapt the template to cross-compile.

Regarding the enabling the new WoW64 for musl only for now, I think that that's a great idea. musl indeed never had support for 32 bit binaries so it would be a great benefit for musl users.

**Comment by fvalasiad** [comment]
@Hoshpak 'Ight let's make this patch merged for now and we shall focus on the other two on a separate PR.

wine for now requires an external emulator for the whole x86 to arm thing, so it'd require extra dependencies for the arm version.

I will also soon figure out how to build musl with new WoW64 and test it with a win32 app, then open a PR.

**Comment by fvalasiad** [comment]
@Hoshpak all checks passed! Feel free to merge!

**Comment by fvalasiad** [comment]
@Hoshpak since you did not merge yet and I finished my experiment with new WoW64 on a musl VM, verifying it works, I included the change in the PR.

Uncertain though if XBPS_LIBC is the correct check to be making here.

**Review by Hoshpak — APPROVED** [review]

**Pull request merged**